<launch>
    <arg name="namespace" default="ns1" />
    <arg name="droneip" default="192.168.1.1" />
    <group ns="$(arg namespace)">
     <node pkg="autonomy_human" name="autonomy_human" type="autonomy_human" output="screen" cwd="node" >
        <param name="cascade_file" value="$(find autonomy_human)/cascades/haarcascade_frontalface_alt.xml" />
        <param name="cascade_profile_file" value="$(find autonomy_human)/cascades/haarcascade_profileface.xml" />
        <param name="profile_hack_enabled" value="false" />
        <param name="skin_enabled" value="false" />
        <param name="gesture_enabled" value="false" />
        <param name="initial_min_score" value="2" />
        <param name="initial_detect_frames" value="6" />
        <param name="initial_reject_frames" value="6" />
        <param name="min_flow" value="10" />
        <param name="min_face_width" value="2" />
        <param name="min_face_height" value="3" />
        <param name="max_face_width" value="90" />
        <param name="max_face_height" value="120" />
        <param name="debug_mode" value="2" />
        <remap from="input_rgb_image" to="image_raw" />
      </node>
      <node pkg="image_view" name="rawimage" type="image_view">
        <param name="autosize" value="true" />
        <remap from="image" to="image_raw" />
      </node>
      <node pkg="image_view" name="debugimage" type="image_view">
        <param name="autosize" value="true" />
        <remap from="image" to="output_rgb_debug" />
      </node>
      <node pkg="autonomy_human" name="get_elected" type="get_elected_demo" output="screen" cwd="node">
        <param name="myName" value="$(arg namespace)"/>
      </node>
<!--
      <node pkg="uvc_camera" type="camera_node" name="uvc_camera" output="screen">
        <param name="width" type="int" value="320" />
        <param name="height" type="int" value="240" />
        <param name="fps" type="int" value="30" />
        <param name="frame" type="string" value="wide_stereo" />
        <param name="device" type="string" value="/dev/video0" />
        <param name="camera_info_url" type="string" value="file://$(find uvc_camera)/example.yaml" />
      </node>
-->
      <node pkg="ardrone_autonomy" name="ardrone_driver" type="ardrone_driver" output="screen" args="-ip $(arg droneip)">
	    <param name="max_bitrate" value="2000" />
            <param name="bitrate" value="2000" />
            <param name="video_codec" value="129" />
            <param name="root_frame" value="0" />
            <param name="do_imu_caliberation" value="true" />
            <param name="outdoor" value="0" />
            <param name="flight_without_shell" value="0" />
            <param name="altitude_max" value="3000" />
            <param name="altitude_min" value="50" />
            <param name="navdata_demo" value="1" />
            <param name="control_vz_max" value="700" />
            <param name="realtime_navdata" value="True" />
            <param name="realtime_video" value="True" />
            <param name="looprate" value="30" />
 
        <remap from="ardrone/image_raw" to="image_raw" />
      </node>
    </group>
</launch>
